syntax = 'proto3';
service TodoService {
    
    rpc CreateTodo (Todo) returns (Todo) {};
    
    rpc UpdateTodo (Todo) returns (Todo) {};

    rpc GetAllTodos (EmptyRequest) returns (Todos){};
    
    rpc GetTodo (Todo) returns (Todo) {};

    rpc DeleteTodo(Todo)returns (EmptyRequest) {};
  }
  
  
message Todo {
  int32 id = 1;
  string todo_text = 2;
  int64 created_at = 3;
  enum Status{
    INPROGRESS = 0;
    COMPLETED = 1;
    CREATED = 2;
  };
  Status status = 4;
}



message Todos{
  repeated Todo todos = 1;
}

message EmptyRequest{

}

  